### ISSUE TITLE
Undefined symbols

### ISSUE THREAD

**Issue by gegles** [gabime/spdlog#660]
Up until recently, everything worked fine for me using spdlog (header-only). i am not doing anything specific. Everything still works for me with 461b5ef28a2afa3e94a1d03fb2c98ae9f4423dcc (from March 9th).

If I update to the latest, I get a ton of Unresolved symbols:

```
Undefined symbols for architecture x86_64:
  "fmt::FormatError::~FormatError()", referenced from:
      fmt::BasicFormatter<char, fmt::ArgFormatter<char> >::format(fmt::BasicCStringRef<char>) 
     ....
```

Am I missing something? Is this a known issue?

**Comment by gabime** [comment]
Please specify which compiler and  version.

**Comment by gegles** [comment]
I am using AppleClang 9.0.0.9000039 but I get the same issue with VS2017:
```
error LNK2019: unresolved external symbol "public: virtual __cdecl fmt::FormatError::~FormatError(void)"
```
I could try with gcc, but I bet it will be the same....

**Comment by gabime** [comment]
include from which file? why did it break in last release?

**Comment by vitaut** [comment]
This happens because the {fmt} headers are included with conflicting macros. spdlog defines `FMT_HEADER_ONLY` in https://github.com/gabime/spdlog/blob/master/include/spdlog/fmt/fmt.h#L16 while when you include `spdlog/fmt/ostr.h` manually first `FMT_HEADER_ONLY` is not defined. It might be a good idea to define `FMT_HEADER_ONLY` in `spdlog/fmt/ostr.h` if this header is supposed to be used by users.

**Comment by gegles** [comment]
Thanks @vitaut! That explains it. Please let me know if I am not supposed to `#include <spdlog/fmt/ostr.h>` directly, as a user, and what I should include for the `operator<<` formatting to work.

**Comment by vitaut** [comment]
@gabime can clarify whether `spdlog/fmt/ostr.h` is intended for direct use. If it is, then I recommend defining `FMT_HEADER_ONLY` in it, possibly via the `spdlog/fmt/fmt.h` include.

**Comment by gabime** [comment]
@vitaut Sounds right - I will add a define FMT_HEADER_ONLY in spdlog/fmt/fmt.h (unless SPDLOG_FMT_EXTERNAL is defined)

**Comment by gabime** [comment]
Fixed in ea95ea8295ecfe43011ff1997cfd6024996c5418
